app/controllers/building_map_controller.rb

```diff
@@ -1,7 +1,7 @@
 class BuildingMapController < ApplicationController
   def index
-    @start = params[:start]
-    @destination = params[:dest]
-    @route = RoutingHelper.calculate_route(@start, @destination)
+    @start = RoutingHelper.valid_coordinates(params[:start]) ? params[:start] : nil
```

Comment:
These can all be shortened like:
```suggestion
    @start = params[:start] if RoutingHelper.valid_coordinates(params[:start])
```